src/org/parosproxy/paros/network/HttpRequestHeader.java

```diff
@@ -449,29 +449,30 @@ private void parse(boolean isSecure) throws URIException, HttpMalformedHeaderExc
             throw new HttpMalformedHeaderException("Unexpected version: " + mVersion);
         }
 
-        mUri = parseURI(sUri);
-
-        if (mUri.getScheme() == null || mUri.getScheme().equals("")) {
-            mUri = new URI(HTTP + "://" + getHeader(HOST) + mUri.toString(), true);
-        }
-
-        if (isSecure() && mUri.getScheme().equalsIgnoreCase(HTTP)) {
-            mUri = new URI(mUri.toString().replaceFirst(HTTP, HTTPS), true);
-        }
-
-        if (mUri.getScheme().equalsIgnoreCase(HTTPS)) {
-            setSecure(true);
-        }
-
-        String hostHeader;
         if (mMethod.equalsIgnoreCase(CONNECT)) {
+            String hostHeader;
             hostHeader = sUri;

```

Comment:
Just do this on one line? `String hostHeader = sUri;`

Reply:
Sorry, I didn't notice this one. Fixing...

Reply:
The `hostHeader` can be removed and `sUri` used directly though.

Reply:
DOH, sorry I didn't pick that up.

Reply:
Thanks for the heads up! Fixed!